sdk/core/core/inc/az_http_response_parser.h

```diff
@@ -90,6 +90,14 @@ az_http_response_get_next_header(az_span const self, az_pair * const p_header);
 AZ_NODISCARD az_result
 az_http_response_get_body(az_span const self, az_pair * const p_last_header, az_span * const body);
 
+/**
+ * Get an HTTP header by name.
+ */
+AZ_NODISCARD az_result az_http_response_get_header_by_name(
```

Comment:
This could be a generic function used by both requests and responses.  az_http_get_header_by_name(.....)

Reply:
I know but we don't have a parser for response. I know that they are almost the same but I don't want to solve this problem now.

Reply:
request and response are internally different. 
Request builder is already exposing a way to iterate headers.  

In that case, we would be introducing some command pattern to ask someone to get header from request or response and let it handle it for us...  some kind of Header_Assistant :)